File: PostApp/app/src/main/java/hootor/com/postapp/service/RetrofitInstance.java
package hootor.com.postapp.service;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitInstance {

    private static Retrofit retrofit= null;
    private static final String BASE_URL = "https://jsonplaceholder.typicode.com/";

    private static Object BLOK = new Object();

    public static PostAppService getService() {

        synchronized (BLOK) {
            if (retrofit == null) {
                retrofit = new Retrofit
                        .Builder()
                        .baseUrl(BASE_URL)
                        .addConverterFactory(GsonConverterFactory.create())
                        .build();
            }
        }

        return retrofit.create(PostAppService.class);
    }
}

File: RetrofitDemo/app/src/main/java/com/android/hootr/retrofitdemo/service/RetrofitInstance.kt
package com.android.hootr.retrofitdemo.service

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class RetrofitInstance private constructor() {

    companion object {

        val BASE_URL = "http://groupkt.com/"

        private var service: CountryService? = null

        fun getService(): CountryService {

            service?.let {
                return it
            }

            val retrofit = Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build()

            val instance = retrofit.create(CountryService::class.java)
            service = instance

            return instance

        }


    }

}